drivers/serial/uart_lpc11u6x.c

```diff
@@ -0,0 +1,811 @@
+/*
+ * Copyright (c) 2020, Seagate Technology LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT nxp_lpc11u6x_uart
+
+#include <arch/arm/aarch32/cortex_m/cmsis.h>
+
+#include <drivers/uart.h>
+#include <drivers/pinmux.h>
+#include <drivers/clock_control.h>
+
+#include "uart_lpc11u6x.h"
+
+#define DEV_CFG(dev) ((dev)->config_info)
+#define DEV_DATA(dev) ((dev)->driver_data)
+
+#ifdef CONFIG_UART_LPC11U6X_PORT0
+static int lpc11u6x_uart0_poll_in(struct device *dev, unsigned char *c)
+{
+	const struct lpc11u6x_uart0_config *cfg = DEV_CFG(dev);
+
+	if (!(cfg->uart0->lsr & LPC11U6X_UART0_LSR_RDR)) {
+		return -1;
+	}
+	*c = cfg->uart0->rbr;
+
+	return 0;
+}
+
+static void lpc11u6x_uart0_poll_out(struct device *dev, unsigned char c)
+{
+	const struct lpc11u6x_uart0_config *cfg = DEV_CFG(dev);
+
+	while (!(cfg->uart0->lsr & LPC11U6X_UART0_LSR_THRE)) {
+	}
+	cfg->uart0->thr = c;
+}
+
+static int lpc11u6x_uart0_err_check(struct device *dev)
+{
+	const struct lpc11u6x_uart0_config *cfg = DEV_CFG(dev);
+	u32_t lsr;
+	int ret = 0;
+
+	lsr = cfg->uart0->lsr;
+	if (lsr & LPC11U6X_UART0_LSR_OE) {
+		ret |= UART_ERROR_OVERRUN;
+	}
+	if (lsr & LPC11U6X_UART0_LSR_PE) {
+		ret |= UART_ERROR_PARITY;
+	}
+	if (lsr & LPC11U6X_UART0_LSR_FE) {
+		ret |= UART_ERROR_FRAMING;
+	}
+	if (lsr & LPC11U6X_UART0_LSR_BI) {
+		ret |= UART_BREAK;
+	}
+
+	return ret;
+}
+
+static int lpc11u6x_uart0_configure(struct device *dev,
+				    const struct uart_config *cfg)
+{
+	return -ENOTSUP;
```

Comment:
Wont this give problems with portable code, which attempts to configure the UART? Would it be better to check the given configuration and only return `-ENOTSUP` if the requested configuration is not applicable for the LPC11U6x?

Reply:
Indeed, I did not think of that. I implemented the configuration functions for values that are supported by LPC11U6X